### PR TITLE
Fix local build creating directories owned by root

### PR DESCRIPTION
**Pull request by mluukkainen**
@mluukkainen in the future please always write a PR description.

### CODE REVIEW

File: ci/local/build.sh [hunk 1]
```diff
@@ -90,6 +91,10 @@ fi
 mkdir -p ${BASE_CONTAINER_BUILD_DIR}
 mkdir -p ${CPP_CONTAINER_BUILD_DIR}
 mkdir -p ${PYTHON_CONTAINER_BUILD_DIR}
+# Create build directories. This is to ensure correct owner for directories. If
+# directories don't exits docker volume mounting side effect is that build directorier would be owned by root
+mkdir -p "${REPO_PATH}/build"
```
Comment (rlratzel) [inline ci/local/build.sh]: you should use `${CPP_BUILD_DIR}` and `${PYTHON_BUILD_DIR}` for consistency.

Comment (mluukkainen) [reply ci/local/build.sh]: Fixed on latest iteration. Uses
mkdir -p "${REPO_PATH}/${CPP_BUILD_DIR}"
mkdir -p "${REPO_PATH}/${PYTHON_BUILD_DIR}"

File: ci/local/build.sh [hunk 2]
```diff
@@ -67,38 +67,71 @@ BASE_CONTAINER_BUILD_DIR=${REPO_PATH}/build_$(echo $(basename ${DOCKER_IMAGE})|s
...
-cd ${WORKSPACE}
+cd \${WORKSPACE}
 if [ -f \${PREBUILD_SCRIPT} ]; then
     source \${PREBUILD_SCRIPT}
 fi
 yes | source \${BUILD_SCRIPT}
 "
 
-if (( ${CONTAINER_SHELL_ONLY} == 0 )); then
+if (( CONTAINER_SHELL_ONLY == 0 )); then
```
Comment (mluukkainen) [inline ci/local/build.sh]: This is done because of: https://github.com/koalaman/shellcheck/wiki/SC2004 $/${} is unnecessary on arithmetic variables.

**Comment by mluukkainen** [comment]
Good to review/merge from my POV

**Comment by mluukkainen** [comment]
@rlratzel Please review/Help assign to someone who can review

**Comment by mluukkainen** [comment]
@rlratzel Rebased on top of branch-0.10.All checks passed. Don't see any root generated files locally. Good to review/merge from my POV.

**Pull request merged**